### PR TITLE
Document workaround for Stapler data binding bug in repeatableProperty

### PR DESCRIPTION
**Pull request by uhafner**
Improve documentation of Stapler `repeatableProperty` so workaround for Stapler data binding bug will be described.

See [dev-list discussion](https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/jenkinsci-dev/T_N8R9LAN-c/yxyt3myqAwAJ).

### Proposed changelog entries

Not required.

### Submitter checklist

- [ ] JIRA issue is well described
- [ ] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@jglick

### CODE REVIEW

**Review by jvz — APPROVED** [review]
I've been hit by this bug before myself. Thanks for the docs update!

File: core/src/main/resources/lib/form/repeatableProperty.jelly [hunk 1]
```diff
@@ -35,6 +35,12 @@ THE SOFTWARE.
             <f:repeatableDeleteButton />
           </div>
         </f:block>
+
+    Due to a bug in Stapler data binding the model elements are only set if they consist of one or more values.
+    If all values have been removed in the user interface (i.e. the associated form is empty), then the setter is not
+    invoked anymore. As a workaround, you need to override the corresponding configure method and clear the model
+    property manually before invoking the data binding. See warnings-ng-plugin PR#266.
```
Comment (timja) [inline core/src/main/resources/lib/form/repeatableProperty.jelly]: is it worth linking directly to the PR if someone actually wants to see the detail?

Comment (uhafner) [reply core/src/main/resources/lib/form/repeatableProperty.jelly]: I'm not sure where this documentation is shown? I always use it as it is in IntelliJ source code view.

Comment (timja) [reply core/src/main/resources/lib/form/repeatableProperty.jelly]: It's published here:
https://reports.jenkins.io/core-taglib/jelly-taglib-ref.html#form:repeatableHeteroProperty
But I assume mostly consumed in IDEs

also see:
https://jenkins.io/doc/developer/development-environment/taglibs/

Comment (uhafner) [reply core/src/main/resources/lib/form/repeatableProperty.jelly]: Hmm, seems that I can't use links there...

**Review by timja — APPROVED** [review]

**Review by varyvol — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
@MarkEWaite I think we need to talk about the review/merge process here. I will try to document it ASAP, but commonly we do the following things:

* Update changelog labels before merging so that release drafter and changelog generator properly draft the changelog
* Put `ready-for-merge` label and notify that the change is going to be merged soon if no negative feedback (e.g. in 24hrs). This will be automated later

**Comment by MarkEWaite** [comment]
Sorry about that @oleg-nenashev!

Can you guide on the changelog labels that would be used for this pull request?  Specifically, there isn't a documentation changelog label, so I didn't know which changelog label to assign to it.  I guess it could be called "developer" or "internal" but I wasn't clear either of those were a good fit for this change.

I'll definitely follow the `ready-to-merge` process for future merge processes.  Thanks for catching my mistake!

**Comment by oleg-nenashev** [comment]
No worries. I will document the process once I find some time for it

**Pull request merged**